packages/runtime-dom/src/modules/events.ts

```diff
@@ -152,3 +152,192 @@ function patchStopImmediatePropagation(
     return value
   }
 }
+
+export interface SyntheticEvent<T = Element, E = Event> extends Event {
+  target: EventTarget
+  nativeEvent: E
+  currentTarget: T & EventTarget
```

Comment:
Thought: From the docs, it says that [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element) is already extending `EventTarget`, do we need `T & EventTarget`?

Reply:
Thought: Maybe we need to have `T extends Element = Element` instead?! 🤔